src/expressions.md

```diff
@@ -214,43 +217,143 @@ assert_eq!(5, x);
 ## Method-call expressions
 
 A _method call_ consists of an expression followed by a single dot, an
-identifier, and a parenthesized expression-list. Method calls are resolved to
-methods on specific traits, either statically dispatching to a method if the
-exact `self`-type of the left-hand-side is known, or dynamically dispatching if
-the left-hand-side expression is an indirect [trait
-object](types.html#trait-objects).
+[identifier](identifiers.html), and a parenthesized expression-list. Method
+calls are resolved to methods on specific traits, either statically dispatching
+to a method if the exact `self`-type of the left-hand-side is known, or
+dynamically dispatching if the left-hand-side expression is an indirect [trait
+object](types.html#trait-objects). If the method was defined to take `&self` or
+`&mut self` then a borrow automatically be taken.
+
+When resolving method calls on an expression of type `A`, Rust will use the
+following order:
+1. Inherent methods, with receiver `A`, `&A`, `&mut A`.
+1. Trait methods with receiver of type `A`.
+1. Trait methods with receiver of type `&A`.
+1. Trait methods with receiver of type `&mut A`.
+1. If it's possible, Rust will then repeat steps 1-5 with
+  `<A as std::ops::Deref>::Target`, and insert a dereference operator.
```

Comment:
This might not be perfect, but it's also not incorrect; we may want to elaborate later, but I don't think it needs to block this PR